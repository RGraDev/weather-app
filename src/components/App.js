import "../styles/App.css";
import LocationDetails from "./LocationDetails";
import ForecastSummaries from "./ForecastSummaries";
import ForecastDetails from "./ForecastDetails";

function App(props) {
  const { forecasts, location } = props;

  return (
    <div className="weather-app">
      <LocationDetails city={location.city} country={location.country} />
      <ForecastDetails forecast={forecasts[0]} />
      <ForecastSummaries forecasts={forecasts} />
    </div>
  );
}

export default App;
