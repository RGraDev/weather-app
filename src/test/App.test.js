import React from "react";
import { render } from "@testing-library/react";
import App from "../components/App";
import { location, forecasts } from "../data/forecast.json";

describe("App", () => {
  test("renders App component correctly", () => {
    const { asFragment } = render(
      <App forecasts={forecasts} location={location} />,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
