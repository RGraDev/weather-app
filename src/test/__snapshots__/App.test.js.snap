// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders App component correctly 1`] = `
<DocumentFragment>
  <div
    class="weather-app"
  >
    <h1>
      Manchester, UK
    </h1>
    <div
      class="forecast=details"
    >
      <div
        class="forecast-details__date"
      >
        Mon Apr 30 2018
      </div>
      <div
        class="forecast-details__description"
      >
        Clear
      </div>
      <div
        class="forecast-details__icon"
      >
        <img
          alt="weather icon"
          src="https://openweathermap.org/img/wn/04d@2x.png"
        />
      </div>
      <div
        class="forecast-details__min-temp"
      >
        4°C
      </div>
      <div
        class="forecast-details__max-temp"
      >
        11°C
      </div>
      <div
        class="forecast-details__humidity"
      >
        30
      </div>
      <div
        class="forecast-details__wind-speed"
      >
        13mph
      </div>
      <div
        class="forecast-details__wind-direction"
      >
        S
      </div>
    </div>
    <div
      class="forecast-summaries"
    >
      <div
        class="forecast-summary"
        data-testid="forecast-summary"
      >
        <div
          class="forecast-summary__date"
        >
          Mon Apr 30 2018
        </div>
        <div
          class="forecast-summary__description"
        >
          Clear
        </div>
        <div
          class="forecast-summary__icon"
          data-testid="forecast-icon"
        >
          <img
            alt="weather icon"
            src="https://openweathermap.org/img/wn/04d@2x.png"
          />
        </div>
        <div
          class="forecast-summary__temperature"
        >
          11°C
        </div>
      </div>
      <div
        class="forecast-summary"
        data-testid="forecast-summary"
      >
        <div
          class="forecast-summary__date"
        >
          Tue May 01 2018
        </div>
        <div
          class="forecast-summary__description"
        >
          Stormy
        </div>
        <div
          class="forecast-summary__icon"
          data-testid="forecast-icon"
        >
          <img
            alt="weather icon"
            src="https://openweathermap.org/img/wn/11d@2x.png"
          />
        </div>
        <div
          class="forecast-summary__temperature"
        >
          13°C
        </div>
      </div>
      <div
        class="forecast-summary"
        data-testid="forecast-summary"
      >
        <div
          class="forecast-summary__date"
        >
          Wed May 02 2018
        </div>
        <div
          class="forecast-summary__description"
        >
          Heavy Snow
        </div>
        <div
          class="forecast-summary__icon"
          data-testid="forecast-icon"
        >
          <img
            alt="weather icon"
            src="https://openweathermap.org/img/wn/13d@2x.png"
          />
        </div>
        <div
          class="forecast-summary__temperature"
        >
          1°C
        </div>
      </div>
      <div
        class="forecast-summary"
        data-testid="forecast-summary"
      >
        <div
          class="forecast-summary__date"
        >
          Thu May 03 2018
        </div>
        <div
          class="forecast-summary__description"
        >
          Tornado
        </div>
        <div
          class="forecast-summary__icon"
          data-testid="forecast-icon"
        >
          <img
            alt="weather icon"
            src="https://openweathermap.org/img/wn/50d@2x.png"
          />
        </div>
        <div
          class="forecast-summary__temperature"
        >
          20°C
        </div>
      </div>
      <div
        class="forecast-summary"
        data-testid="forecast-summary"
      >
        <div
          class="forecast-summary__date"
        >
          Fri May 04 2018
        </div>
        <div
          class="forecast-summary__description"
        >
          Hazy
        </div>
        <div
          class="forecast-summary__icon"
          data-testid="forecast-icon"
        >
          <img
            alt="weather icon"
            src="https://openweathermap.org/img/wn/50d@2x.png"
          />
        </div>
        <div
          class="forecast-summary__temperature"
        >
          25°C
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
